src/canvas.ts

```diff
@@ -218,25 +223,126 @@ export class CanvasAssignmentWrapper {
         if(this.canvas_general_infos.description){
             desc+=this.canvas_general_infos.description;
         } else if (this.canvas_assignment.description){
             desc+=this.canvas_assignment.description;
         }
+        //If this assignment has a description we will add one new line character before pasting the link to the canvas
+        // assignment, otherwise we dont add extra white space
         if(desc){
             desc+="\n"
         }
+        //link the canvas assignment itself, allowing users to redirect to canvas to turn in things
         if (this.canvas_assignment.html_url){
             desc += this.canvas_assignment.html_url;
         }
         return desc
     }
 }
 
+export class CanvasEventWrapper {
+    canvas_event: CanvasCalendarEvent;
+    canvas_course_name: string;
+
+    constructor(canvas_course:string, canvas_assignment: CanvasCalendarEvent) {
+        this.canvas_event = canvas_assignment;
+        this.canvas_course_name = canvas_course;
+    }
+
+    get start_date(): Date {
+        //if this is an all day event lets set the start time to just after midnight, beginning of the day
+        if(this.canvas_event.all_day){
+            let date = new Date(this.canvas_event.all_day_date)
+            date.setHours(0,0,0,0)
+            return date
+        }
+        return new Date(
+            //begins
+            this.canvas_event.start_at
+        );
+    }
+
+    get end_date(): Date {
+        //if this event is an all day one lets send the end time to just before midnight, the end of the day
+        if(this.canvas_event.all_day){
+            let date = new Date(this.canvas_event.all_day_date)
+            date.setHours(23,59,59,999);
+            return date;
+        }
+        return new Date(
+            //begins
+            this.canvas_event.end_at
+        );
+    }
+    /** Returns short description of a canvas assignment with the specified format
+     *  @returns: String of pattern [COURSE NAME CODE]: Assignment Name.
+     * */
+    get title(): string {
+        let desc = "";
+        //Lets start by getting the course name and a colon
+        if (this.canvas_course_name){
+            desc += "" + this.canvas_course_name.toLowerCase().trim()
+        }
+        else if (this.canvas_event.context_name){
+            desc += "" + this.canvas_event.context_name.toLowerCase().trim();
+        } else {
+            console.log("ERROR");
+            desc += "COURSE MISSING";
+        }
+
+        //lets now get the assignment name and remove duplicate words from the course name
+        if(this.canvas_event.title && this.canvas_event.title.toLowerCase().trim() !== desc){
+            let comparison = desc.split(" ");
+            desc += " : ";
+            //to clean we split the assignment name by spaces, then only include words NOT in the course name
+             let cleaned_name = this.canvas_event.title.toLowerCase()
```

Comment:
Isn't this going to make event titles all lowercase? We wouldn't really want this right?